internal/iam/repository_role_grant_test.go

```diff
@@ -2429,3 +2429,77 @@ func SetupChildGrantScopes(t *testing.T, conn *db.DB, repo *Repository) (childGr
 	)
 	return
 }
+
+func TestGrantsForUserGlobalResources(t *testing.T) {
+	ctx := context.Background()
+	conn, _ := db.TestSetup(t, "postgres")
+	wrap := db.TestWrapper(t)
+	repo := TestRepo(t, conn, wrap)
+	user := TestUser(t, repo, "global")
+
+	// Create scopes
+	org1 := TestOrg(t, repo)
+	org2 := TestOrg(t, repo)
+
+	// Create roles
+	role1 := TestRole(t, conn, globals.GlobalPrefix, WithGrantScopeIds([]string{org1.PublicId}))
+	role2 := TestRole(t, conn, globals.GlobalPrefix, WithGrantScopeIds([]string{globals.GrantScopeThis, org2.PublicId}))
+	role3 := TestRole(t, conn, globals.GlobalPrefix, WithGrantScopeIds([]string{globals.GrantScopeDescendants}))
+	role4 := TestRole(t, conn, globals.GlobalPrefix, WithGrantScopeIds([]string{globals.GrantScopeThis, globals.GrantScopeChildren}))
+
+	// Grant roles
+	TestRoleGrant(t, conn, role1.PublicId, "ids=*;type=alias;actions=create,update,read,list")
+	TestRoleGrant(t, conn, role1.PublicId, "ids=*;type=alias;actions=delete")
+	TestRoleGrant(t, conn, role2.PublicId, "ids=*;type=alias;actions=delete")
+	TestRoleGrant(t, conn, role3.PublicId, "ids=*;type=*;actions=update")
+	TestRoleGrant(t, conn, role4.PublicId, "ids=*;type=account;actions=create,update")
+	TestRoleGrant(t, conn, role4.PublicId, "ids=*;type=group;actions=read;output_fields=id")
+
+	// Add user to created roles
+	for _, roleId := range []string{role1.PublicId, role2.PublicId, role3.PublicId, role4.PublicId} {
+		_, err := repo.AddPrincipalRoles(ctx, roleId, 1, []string{user.PublicId})
+		require.NoError(t, err)
+	}
+
+	t.Run("Alias", func(t *testing.T) {
+		got, err := repo.grantsForUserGlobalResources(ctx, user.PublicId, resource.Alias)
```

Comment:
Can we expand on these tests? It would be good to cover other cases including:
- error cases
- multiple grant tuples being returned
- grants for the special `u_auth` and `u_anon` users

Reply:
Added more test coverage: 60181c3c287a3f3947c54ee174b9fba4b03237f1